src/engine/client/key_identification.cpp

```diff
@@ -0,0 +1,504 @@
+	{ "XBOX360_X",              keyNum_t::K_XBOX360_X              },
+	{ "XBOX360_Y",              keyNum_t::K_XBOX360_Y              },
+	{ "XBOX360_LB",             keyNum_t::K_XBOX360_LB             },
+	{ "XBOX360_RB",             keyNum_t::K_XBOX360_RB             },
+	{ "XBOX360_START",          keyNum_t::K_XBOX360_START          },
+	{ "XBOX360_GUIDE",          keyNum_t::K_XBOX360_GUIDE          },
+	{ "XBOX360_LS",             keyNum_t::K_XBOX360_LS             },
+	{ "XBOX360_RS",             keyNum_t::K_XBOX360_RS             },
+	{ "XBOX360_BACK",           keyNum_t::K_XBOX360_BACK           },
+	{ "XBOX360_LT",             keyNum_t::K_XBOX360_LT             },
+	{ "XBOX360_RT",             keyNum_t::K_XBOX360_RT             },
+	{ "XBOX360_DPAD_UP",        keyNum_t::K_XBOX360_DPAD_UP        },
+	{ "XBOX360_DPAD_RIGHT",     keyNum_t::K_XBOX360_DPAD_RIGHT     },
+	{ "XBOX360_DPAD_DOWN",      keyNum_t::K_XBOX360_DPAD_DOWN      },
+	{ "XBOX360_DPAD_LEFT",      keyNum_t::K_XBOX360_DPAD_LEFT      },
+	{ "XBOX360_DPAD_RIGHTUP",   keyNum_t::K_XBOX360_DPAD_RIGHTUP   },
+	{ "XBOX360_DPAD_RIGHTDOWN", keyNum_t::K_XBOX360_DPAD_RIGHTDOWN },
+	{ "XBOX360_DPAD_LEFTUP",    keyNum_t::K_XBOX360_DPAD_LEFTUP    },
+	{ "XBOX360_DPAD_LEFTDOWN",  keyNum_t::K_XBOX360_DPAD_LEFTDOWN  },
+
+	{ nullptr,                  Key::NONE                          }
+};
+
+static const char CHARACTER_BIND_PREFIX[] = "char:";
+static const char SCANCODE_ASCII_BIND_PREFIX[] = "hw:";
+
+static int ParseCharacter(Str::StringRef s)
+{
+    if (!Q_stricmp(s.data(), "SEMICOLON"))
+        return ';';
+    if (!Q_stricmp(s.data(), "BACKSLASH"))
+        return '\\';
+    if (!Q_stricmp(s.data(), "SPACE"))
+        return ' ';
+    if (Q_UTF8_Strlen(s.data()) == 1)
+        return Str::ctolower(Str::UTF8To32(s.data())[0]);
+    return 0;
+}
+
+// This takes a character name and tries to find a key that types that character.
+// If the SDL lookup function doesn't find one, fall back to the scancode
+// for the key on a QWERTY layout, if possible.
+static Key KeyFromUnprefixedCharacter(int ch)
+{
+    SDL_Scancode sc = SDL_GetScancodeFromKey(Util::enum_cast<SDL_Keycode>(ch));
+    if (sc != SDL_SCANCODE_UNKNOWN) {
+        return Key::FromScancode(sc);
+    }
+    return Key::FromScancode(AsciiToScancode(ch));
+}
+
+/*
+===================
+Key_StringToKeynum
+
+Returns a key number to be used to index keys[] by looking at
+the given string.  Single ascii characters return themselves, while
+the K_* names are matched up.
+
+0x11 will be interpreted as raw hex, which will allow new controlers
+
+to be configured even if they don't have defined names.
+===================
+*/
+Key Key_StringToKeynum( const char *str )
+{
+	const keyname_t *kn;
+
+	if ( !str )
+	{
+		return Key::NONE;
+	}
+
+	if ( int c = ParseCharacter( str ) )
+	{
+		return KeyFromUnprefixedCharacter( c );
+	}
+
+	// check for hex code
+	if ( str[0] == '0' && str[1] == 'x' )
+	{
+		int n = Com_HexStrToInt( str );
```

Comment:
Not really a comment on your commit, but the code for Com_HexStrToInt could be replaced entirely by strtol(char, end, 16)...